### PR TITLE
GH-2298 - File used percentage slightly off

### LINKED ISSUE #2298: File used percentage slightly off
Screen Shot 2020-10-01 at 5 21 12 PM

v.s.

Screen Shot 2020-10-01 at 5 21 43 PM

This is due to the difference in total space calculation.

Lustre does:

`100 * used / (used + available)`

Where IML does:

`100 * (used / total)`

The difference is small, but if it's over a rounding boundary it becomes noticeable.

Lustre:

`100 * 787636048 / (787636048 + 30516274432) == 2.516094749578392`

IML:

`100 * (806539313152 / 32589115686912) == 2.4748732702676906`

If IML does the same calculation as Lustre, we get the same result:

`100 * 806539313152 / (806539313152 + 31248665018368) == 2.516094749578392`

### PR DESCRIPTION
**Pull request by johnsonw**
Update percentage on chart and in retention to match the percentage
displayed by lustre.

Fixes #2298.

Signed-off-by: johnsonw <wjohnson@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2302)

### CODE REVIEW

**Comment by johnsonw** [comment]
In the process of testing this.

**Comment by johnsonw** [comment]
image

image
*Note, I have two filesystems here.*

```
[root@oss1 fs]# lfs df -h
UUID                       bytes        Used   Available Use% Mounted on
fs-MDT0000_UUID             2.8G        2.4M        2.5G   1% /mnt/fs[MDT:0]
fs-MDT0001_UUID             2.8G        2.4M        2.5G   1% /mnt/fs[MDT:1]
fs-OST0000_UUID             4.6G        2.0M        4.3G   1% /mnt/fs[OST:0]
fs-OST0001_UUID             4.6G        2.8G        1.5G  66% /mnt/fs[OST:1]
fs-OST0002_UUID             4.6G      129.9M        4.2G   3% /mnt/fs[OST:2]
fs-OST0003_UUID             4.6G        1.0G        3.3G  24% /mnt/fs[OST:3]
fs-OST0004_UUID             4.6G        2.0M        4.3G   1% /mnt/fs[OST:4]
fs-OST0005_UUID             4.6G        1.0G        3.3G  24% /mnt/fs[OST:5]
fs-OST0006_UUID             4.6G        1.0G        3.3G  24% /mnt/fs[OST:6]
fs-OST0007_UUID             4.6G        2.0M        4.3G   1% /mnt/fs[OST:7]
fs-OST0008_UUID             4.6G        1.0G        3.3G  24% /mnt/fs[OST:8]
fs-OST0009_UUID             4.6G        2.0M        4.3G   1% /mnt/fs[OST:9]

filesystem_summary:        45.8G        7.0G       36.1G  17% /mnt/fs
```

**Comment by johnsonw** [comment]
Also wanted to show that the files chart is still working as well:
image

File: iml-gui/crate/src/components/progress_circle.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ pub fn used_to_color(used: f64) -> &'static str {
 pub fn view<'a, T>(x: impl Into<Option<(f64, &'a str)>>) -> Node<T> {
     match x.into() {
         Some((used, color)) => {
-            let stroke_length = (1.0 - used) * CIRCUMFERENCE;
+            let stroke_length = (100.0f64 - used) / 100.0f64 * CIRCUMFERENCE;
```
Comment (jgrund) [inline iml-gui/crate/src/components/progress_circle.rs:27]: nit: f64 is unnecessary when using a decimal.

**Review by jgrund — APPROVED** [review]

**Comment by ip1981** [comment]
Does CLI have the same issue?

File: iml-gui/crate/src/components/progress_circle.rs (hunk 1 above)
Comment (johnsonw) [reply iml-gui/crate/src/components/progress_circle.rs:27]: Yeah I just thought I would be explicit :)

**Comment by johnsonw** [comment]
@ip1981, I could be missing something but I think it's ok:

```
[root@adm report]# iml filesystem show fs
+-------------------+---------------------------------------+
| Name              | fs                                    |
+-------------------+---------------------------------------+
| Space Used/Avail  | 7 GiB / 36 GiB                        |
+-------------------+---------------------------------------+
| Inodes Used/Avail | 1M / 3M                               |
+-------------------+---------------------------------------+
| State             | available                             |
+-------------------+---------------------------------------+
| Management Server | mds1.local                            |
+-------------------+---------------------------------------+
| MDTs              | fs-MDT0000                            |
|                   | fs-MDT0001                            |
+-------------------+---------------------------------------+
| OSTs              | fs-OST0000                            |
|                   | fs-OST0004                            |
|                   | fs-OST0002                            |
|                   | fs-OST0003                            |
|                   | fs-OST0001                            |
|                   | fs-OST0008                            |
|                   | fs-OST0007                            |
|                   | fs-OST0009                            |
|                   | fs-OST0006                            |
|                   | fs-OST0005                            |
+-------------------+---------------------------------------+
| Clients           | 1                                     |
+-------------------+---------------------------------------+
| Mount Path        | 10.73.20.11@tcp0:10.73.20.12@tcp0:/fs |
+-------------------+---------------------------------------+
```

Let me know if there is a different command you are referring to.

**Review by ip1981 — APPROVED** [review]

**Pull request merged**